File: Algorithms/Java/SumOfDigitsRecursive.java
public class SumOfDigitsRecursive {
    public static void main(final String[] args) {
        int n = 123;
        System.out.println(sumOfDigits(n));
    }

    public static int sumOfDigits(final int number) {
        if (number == 0) {
            return 0;
        }
        return number % 10 + sumOfDigits(number / 10);
    }
}

File: Algorithms/Python/sum_of_digits_recursive.py
from __future__ import annotations


def sum_of_digits_recursive(n: int) -> int:
    """
    Calculate the sum of digits of a number using recursion.
    """
    return 0 if n == 0 else n % 10 + sum_of_digits_recursive(n // 10)


if __name__ == "__main__":
    print(sum_of_digits_recursive(1213))
